packages/react-dom/src/test-utils/FizzTestUtils.js

```diff
@@ -133,6 +139,86 @@ async function replaceScriptsAndMove(
   }
 }
 
+export function installScriptObserver(window: any, document: Document) {
+  let observer = window.__scriptObserver;
+  if (!observer) {
+    window.__observedScripts = new Set();
+    window.__loadScripts = loadScripts;
+    window.eval(scriptObserverFunctionString);
+    observer = window.__scriptObserver;
+  }
+  observer.observe(document, {childList: true});
+  if (document.documentElement) {
+    observer.observe(document.documentElement, {childList: true});
+  }
+  if (document.head) {
+    observer.observe(document.head, {childList: true});
+  }
+  if (document.body) {
+    observer.observe(document.body, {childList: true});
+  }
+}
+
+export function removeScriptObserver(window: any) {
+  if (window.__scriptObserver) {
+    window.__scriptObserver.disconnect();
+  }
+}
+
+const scriptObserverFunctionString = `
+  const scriptObserver = new MutationObserver(mutations => {
+    let promises = [];
+    for (var i = 0; i < mutations.length; i++) {
+      const addedNodes = mutations[i].addedNodes;
+      if (addedNodes.length) {
+        var pendingWork = window.__loadScripts(window);
+        window.__pendingWork = (window.__pendingWork || Promise.resolve()).then(() => pendingWork);
+        // We can stop here because loadScripts loads scripts globally so there
+        // is not additional benefit to calling it for each mutation that added nodes
+      }
+    }
+  });
+
+  window.__scriptObserver = scriptObserver;
+`;
```

Comment:
The reason this is a string and is eval'd is because the MutationObserver does not think the Node's passed in are actually Nodes. This is because `window.Node` is not in the prototype chain in the context of this file since we are using an adhoc jsdom window not the actual global. @mofeiZ if you can think of way to do this so we don't need to have this code as a string I'm open to suggestions :)